src/components/data/EntitiesContent.test.ts

```diff
@@ -0,0 +1,400 @@
+
+jest.mock('src/libs/ajax');
+
+type ReactNotificationsComponentExports = typeof import('react-notifications-component');
+jest.mock('react-notifications-component', (): DeepPartial<ReactNotificationsComponentExports> => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+type ReactVirtualizedExports = typeof import('react-virtualized');
+jest.mock('react-virtualized', (): ReactVirtualizedExports => {
+  const actual = jest.requireActual<ReactVirtualizedExports>('react-virtualized');
+
+  const { AutoSizer } = actual;
+  class MockAutoSizer extends AutoSizer {
+    state = {
+      height: 1000,
+      width: 1000,
+    };
+
+    setState = () => {};
+  }
+
+  return {
+    ...actual,
+    AutoSizer: MockAutoSizer,
+  };
+});
+
+type ClipboardPolyfillExports = typeof import('clipboard-polyfill/text');
+jest.mock('clipboard-polyfill/text', (): ClipboardPolyfillExports => {
+  const actual = jest.requireActual<ClipboardPolyfillExports>('clipboard-polyfill/text');
+  return {
+    ...actual,
+    writeText: jest.fn().mockResolvedValue(undefined),
+  };
+});
+
+type FileSaveExports = typeof import('file-saver');
+jest.mock('file-saver', (): FileSaveExports => {
+  const actual = jest.requireActual<FileSaveExports>('file-saver');
+  return {
+    ...actual,
+    saveAs: jest.fn().mockResolvedValue(undefined),
+  };
+});
+
+describe('EntitiesContent', () => {
+  it('copies to clipboard', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    const paginatedEntitiesOfType = jest.fn().mockResolvedValue({
+      results: [
+        {
+          entityType: 'sample',
+          name: 'sample_1',
+          attributes: {},
+        },
+      ],
+      resultMetadata: { filteredCount: 1, unfilteredCount: 1 },
+    });
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Workspaces: {
+        workspace: () => {
+          return {
+            paginatedEntitiesOfType,
+          };
+        },
+      },
+      Metrics: {
+        captureEvent: () => {},
+      },
+    };
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+    await act(async () => {
+      render(
+        h(EntitiesContent, {
+          workspace: {
+            ...defaultGoogleWorkspace,
+            workspace: {
+              ...defaultGoogleWorkspace.workspace,
+              attributes: {},
+            },
+            workspaceSubmissionStats: {
+              runningSubmissionsCount: 0,
+            },
+          },
+          entityKey: 'sample',
+          activeCrossTableTextFilter: '',
+          entityMetadata: {
+            sample: {
+              idName: 'sample_id',
+              attributeNames: [],
+              count: 1,
+            },
+          },
+          setEntityMetadata: () => {},
+          loadMetadata: () => {},
+          snapshotName: null,
+          deleteColumnUpdateMetadata: () => {},
+        })
+      );
+    });
+
+    // Act
```

Comment:
Possibly this is a convention that's in the code, so take with a grain of salt... but given the method call right above this comment is called `act` you could drop the `// Act` comment throughout and just assume the act portion of the test begins when `act` is invoked?

Reply:
This is a pattern that @msilva-broad encouraged, explicitly identifying and separating the "arrange", "act", and "assert" sections of each test. It does look particularly repetitive when the "act" section is wrapped in the `act` function, but that will not always be the case (the `act` function is often not needed).